### PR TITLE
Ongoing facehugger fixes.

### PR DESCRIPTION
**Pull request by Ghommie**
:cl:
fix: Ongoing facehugger fixes.
/:cl:

Fixes more edge cases and omissions such as synths getting knocked out, facehuggers possibly dying while attached, potential runtimes, inability to place facehuggers back into eggs, throwing/leaping vars not resetting.

### CODE REVIEW

**Comment by Rohesie** [comment]
Is there a need to changelog this?

File: code/modules/mob/living/carbon/xenomorph/Facehuggers.dm [hunk 1]
```diff
@@ -433,22 +442,19 @@
 	if(ishuman(user))
 		playsound(loc, (user.gender == MALE ?'sound/misc/facehugged_male.ogg' : 'sound/misc/facehugged_female.ogg') , 25, 0)
 	if(!sterile)
-		if(!issynth(user) || !isIPC(user)) //synthetics aren't paralyzed
+		if(!issynth(user) && !isIPC(user)) //synthetics aren't paralyzed
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/Facehuggers.dm]: Why the and here?

File: code/modules/mob/living/carbon/xenomorph/Facehuggers.dm [hunk 2]
```diff
@@ -192,19 +195,19 @@
...
-	var/obj/item/clothing/mask/facehugger/F
-	var/count = 0
-	for(F in get_turf(src))
-		if(F.stat == CONSCIOUS && !F.sterile)
-			count++
-		if(count > 2) //Was 5, our rules got much tighter
-			visible_message("<span class='xenowarning'>The facehugger is furiously cannibalized by the nearby horde of other ones!</span>")
-			qdel(src)
-			return TRUE
+	if(isturf(loc))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/Facehuggers.dm]:
```suggestion
	if(isturf(loc))
```
early return

Comment (Ghommie) [reply code/modules/mob/living/carbon/xenomorph/Facehuggers.dm]: It is still won't grant a TRUE return value even if the location happen to be a turf, plus it's not too deeply indented, hence why I didn't do the early return.

**Pull request merged**